api/models.py

```diff
@@ -89,3 +90,104 @@ def save(self, *args, **kwargs):
     def __str__(self):
         return '{}{}{}'.format(self.item_code, self.serial_number,
                                self.model_number)
+
+
+class UserManager(BaseUserManager):
+    def create_user(self, email, name, cohort, slack_handle,
+                    is_active=True, is_admin=False, is_staff=False,
+                    phone_number=None, picture=None, password=None):
+        if not email:
+            raise ValueError("Email address is required")
+        elif not name:
+            raise ValueError("Name is required")
+        elif not cohort:
+            raise ValueError("Cohort is required")
+        elif not slack_handle:
+            raise ValueError("Slack handle is required")
+
+        user = self.model(
+            email=self.normalize_email(email),
+            name=name,
+            cohort=cohort,
+            slack_handle=slack_handle,
+        )
+
+        user.set_password(password)
+        user.picture = picture
+        user.staff = is_staff
+        user.admin = is_admin
+        user.active = is_active
+        user.phone_number = phone_number
+        user.save(using=self._db)
+        return user
+
+    def create_staffuser(self, email, name, cohort,
+                         slack_handle, password=None):
+        user = self.create_user(
+            email=email,
+            name=name,
+            cohort=cohort,
+            slack_handle=slack_handle,
+            password=password,
+            is_staff=True,
+        )
+        return user
+
+    def create_superuser(self, email, name, cohort,
+                         slack_handle, password=None):
+        user = self.create_user(
+            email=email,
+            name=name,
+            cohort=cohort,
+            slack_handle=slack_handle,
+            password=password,
+            is_staff=True,
+            is_admin=True
+        )
+        return user
+
+
+class User(AbstractBaseUser):
+    email = models.EmailField(max_length=50, unique=True)
+    name = models.CharField(max_length=50, blank=True, null=True)
```

Comment:
i'm thinking we go with first and last name since it's easier to work with that over the full name

Reply:
I agree